### DISCUSSION TITLE
Using Azure AD as auth provider

### DISCUSSION THREAD

**Discussion by arvinoids** [pocketbase/pocketbase#3371] [Q&A] [answered]
Hello,

I enabled Microsoft auth provider and entered the app ID and a newly generated client secret. After login, this error is logged on the console. Not sure what to do since I am fairly certain that the secret is correct. I have even tried regenerating and I get the same error.
This is the error in the pocketbase log:
```
{
  "errorMessage": "oauth2: \"invalid_client\" \"AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app '09a65885-2540-48d8-9a98-9719129b081a'.\\r\\nTrace ID: 48035bfc-48d5-4f1c-9ac4-218ce5fd4d01\\r\\nCorrelation ID: d4f4b149-b84b-454b-b81c-8eb906909c28\\r\\nTimestamp: 2023-09-22 12:28:20Z\" \"https://login.microsoftonline.com/error?code=7000215\""
}
```

Thanks, 
Arvin

**Comment by ganigeorgiev** [comment]
My Microsoft/Azure trial account has expired and I can't verify it right now but double check that you are using the correct client secret value.

You could also check this comment that provides some details how to setup the Microsoft/Azure OAuth2 app - https://github.com/pocketbase/js-sdk/issues/176#issuecomment-1518939458.

Supabase also has a very detailed guide how to obtain Microsoft/Azure OAuth2 app client secret that you may find useful - https://supabase.com/docs/guides/auth/social-login/auth-azure#obtain-a-secret-id (**note that as mentioned in the link you need to copy the "Value" column string not the "Secret ID"**).

**Reply by LickABrick** [reply]
Off-topic and not sure if you are interested in this, but you can [create a free development tenant](https://developer.microsoft.com/en-us/microsoft-365/dev-program). As long as you do something in the tenant it will auto renew every 90 days (sometimes I do nothing and it still renews)

**Comment by arvinoids** [comment, marked as answer]
Thanks, I got it wrong by using the Secret ID instead of the value, and now it is working.

**Comment by ctrachte** [comment]
Struggled with web authentication for a while several errors, but this error was one of them.

My organization does not participate in MPN, and the Microsoft Azure side had some changes recently that don't seem to be reflected in current discussion threads, seems like many options have been rebranded or renamed. 

Lot's of different errors from Pocketbase from all different Azure test implementations. Below is what worked for me for a web app implementation:

Start by logging in to the Azure portal, navigating to Azure>Microsoft Entra ID>Home>Default Directory>App Registrations

- Add a new application, fill in as many details as make sense. Navigate to the dashboard for that app.
 - Configure app as Web Client (NOT SPA, even if your app is an SPA), the code challenge will fail otherwise.
 - Use the appropriate Pocketbase redirect URI ( as mentioned in the Pocketbase authentication documentation)
 - From the `overview` panel in your Microsoft Entra App you can copy/paste your Tenant ID into the auth and token URLs in the  Pocketbase admin console (when configuring Microsoft Authentication form)
 - From the `overview` panel in your Microsoft Entra App (in Azure), use the `Application (client) ID` as the `Client ID` in the Pocketbase admin console (when configuring Microsoft Authentication form)
 - Add a client secret, use the secret value (not ID) in your Pocketbase admin console (when configuring Microsoft Authentication form)
 - under `Token Configuration ` panel, I added only one row for email access
 - Under `API permissions ` I added: `Microsoft Graph> email, offline_access, User.Read, User.ReadAll, openid` (likely not all of these are necessary, may depend on your app)
 - Under `Branding and Properties` I also verified my domain, but I am not sure if this was affecting the auth flow, doesn't seem to matter initially.
 - Under Owner Panel, added myself as the owner (Whoever you logged in as).

**Comment by kerns** [comment]
Anyone running into this error?

> AADSTS50011: The redirect URI 'http://127.0.0.1:8090/api/oauth2-redirect' specified in the request does not match the redirect URIs configured for the application 'fd0bbbff-3314-4a65-9f1a-XXXXXXXXX'. Make sure the redirect URI sent in the request matches one added to your application in the Azure portal. Navigate to https://aka.ms/redirectUriMismatchError to learn more about how to fix this.

Confusing as I have `localhost:8090` configured as the redirect in my app within Azure. 🤔

**Reply by ganigeorgiev** [reply]
_I don't remember if that was the case with Azure but many providers treat the hostname (`localhost`) and IP (`127.0.0.1`) as different URIs, so make sure that the SDK initialization url matches with the one you are using as redirect URI in your Azure OAuth2 app, aka. if you have `new PocketBase("http://127.0.0.1:8090")`, then the OAuth2 callback address should be `http://127.0.0.1:8090/api/oauth2-redirect`_.

**Reply by kerns** [reply]
Thanks, @ganigeorgiev ...FWIW I was unable to set `http://127.0.0.1:8090/api/oauth2-redirect` in my application the first time I think due to the lack of https ? But I will take a second look. 🙏

**Reply by kerns** [reply]
Yep that was the case. 
![Screenshot 2024-06-23 at 16 13 47](https://github.com/pocketbase/pocketbase/assets/20254/c9a19c16-e769-4155-893d-5fa36b4f516e)

**Reply by kerns** [reply]
Sorry – sorry @ganigeorgiev  – the source of my confusion was really the startup of pocketbase... 

![Screenshot 2024-06-23 at 16 17 56](https://github.com/pocketbase/pocketbase/assets/20254/00d29709-c469-47dc-a45c-271686b9cbad)

I know localhost and 127.0.0.1 are essentially the same locally...but I had had it in my head (and my .env) that it was one of pb's quirks that it wanted 127.0.0.1. So I edited my .env to use localhost instead of 127.0.0.1 and it's all working now. 🙏
